### PR TITLE
Fix some `clippy::doc_markdown` lints in masonry.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This is a start on #449.

### CODE REVIEW

File: masonry/src/testing/harness.rs [hunk 1]
```diff
@@ -336,7 +336,7 @@ impl TestHarness {
 
     // --- MARK: EVENT HELPERS ---
 
-    /// Move an internal mouse state, and send a MouseMove event to the window.
+    /// Move an internal mouse state, and send a [`PointerEvent::PointerMove`] event to the window.
```
Comment (DJMcNab) [inline masonry/src/testing/harness.rs:339]:
```suggestion
    /// Move an internal mouse state, and send a [`PointerMove`](PointerEvent::PointerMove) event to the window.
```
might be slightly nicer formatting.

File: masonry/src/text/edit.rs [hunk 2]
```diff
@@ -75,7 +75,7 @@ impl<T: EditableText> TextEditor<T> {
 
     /// Rebuild the text.
     ///
-    /// See also [TextLayout::rebuild](crate::text2::TextLayout::rebuild) for more comprehensive docs.
+    /// See also [`TextLayout::rebuild()`](crate::text::TextLayout::rebuild) for more comprehensive docs.
```
Comment (DJMcNab) [inline masonry/src/text/edit.rs:78]: I'm not really sure I like the function call brackets here. E.g. https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#indexing doesn't use that form, unless it cares about the value of the argument.

**Review by DJMcNab — APPROVED** [review]

File: masonry/src/text/edit.rs (hunk 2 above)
Comment (waywardmonkeys) [reply masonry/src/text/edit.rs:78]: I tend to do them partly out of a habit from places where you might refer to either a member or a function, and from Doxygen which auto-links things that use `()` ... but I have got them into lots of Rust code now!

Comment (Philipp-M) [reply masonry/src/text/edit.rs:78]: I think it makes sense to stay consistent, and I don't know of crates that use `()` for doc-links.

**Comment by waywardmonkeys** [comment]
Updated per comments.

**Pull request merged**